File: DukeMapGenLib/src/java/trn/duke/experiments/SectorPrefab.java
package trn.duke.experiments;

import trn.Sector;

/**
 * Like wall prefab, but for sectors.
 * @author Dave
 *
 */
public class SectorPrefab {

	Integer floorTexture = null;
	Integer ceilingTexture = null;
	Short floorShade = null;
	Short ceilingShade = null;
	
	public SectorPrefab(int floorTex, int ceilingTex){
		this.floorTexture = floorTex;
		this.ceilingTexture = ceilingTex;
	}
	
	public SectorPrefab(SectorPrefab copyMe){
		this.floorTexture = copyMe.floorTexture;
		this.ceilingTexture = copyMe.ceilingTexture;
	}
	
	public SectorPrefab setFloorShade(short floorShade){
		this.floorShade = (short)floorShade;
		return this;
	}
	
	public SectorPrefab setCeilingShade(short ceilingShade){
		this.ceilingShade = ceilingShade;
		return this;
	}
	
	public void writeTo(Sector sector){
		if(floorTexture != null){
			sector.setFloorTexture(floorTexture);
		}
		
		if(floorShade != null){
			sector.setFloorShade(floorShade);
		}
		
		if(ceilingTexture != null){
			sector.setCeilingTexture(ceilingTexture);
		}
		
		if(ceilingShade != null){
			sector.setCeilingShade(ceilingShade);
		}
		
	}
}

File: DukeMapGenLib/src/scala/duchy/experiments/render/maze/SectorPrefab.scala
package duchy.experiments.render.maze

import trn.Sector

class SectorPrefab(floorTexture: Int, ceilingTexture: Int) {
  var floorShade: Option[Short] = None // this was copied from java code
  var ceilingShade: Option[Short] = None

  def setFloorShade(floorShade: Short): SectorPrefab = {
    this.floorShade = Some(floorShade)
    this
  }

  def setCeilingShade(ceilingShade: Short): SectorPrefab = {
    this.ceilingShade = Some(ceilingShade)
    this
  }

  def writeTo(sector: Sector): Unit = {
    sector.setFloorTexture(floorTexture)
    sector.setCeilingTexture(ceilingTexture)
    this.floorShade.foreach(sector.setFloorShade)
    this.ceilingShade.foreach(sector.setCeilingShade)
  }
}
